### PR TITLE
Make sort order consistent in split queries

### LINKED ISSUE #26808: Ensure that split queries with non-deterministic Skip/Take don't produce wrong results (by adding deterministic orderings)
When doing split query, if there's any non-determinism in the query that may produce wrong results, since the multiple queries may return different/inconsistent data.

In some cases we can be sure the query is non-deterministic, and can throw (#21202 tracks that). In other cases (https://github.com/dotnet/EntityFramework.Docs/issues/3242) we may not be sure (e.g. depends on the uniqueness of the last column being ordered by), so throwing may be too much (and block legitimate, non-dangerous use). A warning may be more appropriate here (probably needs to be discussed).

### PR DESCRIPTION
**Pull request by ascott18**
Make sort order consistent in split queries between the parent query and its occurrences in subqueries for collection includes.

- Deterministic sorting was already being added to the parent query (sorting by the identifiers used to join the subquery to the subsequent collection queries). However, due to what appears to be a simple oversight, these were not getting applied to the subquery in the split collection query. As a result, the parent query and the subquery were actively being given **different sort orderings**.
- Tests that previously expected a thrown exception (`Include_collection_skip_take_no_order_by`, `Include_collection_skip_no_order_by`) due to this missing sort order in skip/take queries have been made non-throwing since the same ordering from the parent query is now applied to the subquery that lacked sorting.

All the changes in the tests that add or change an `ORDER BY` are such that the change makes the subquery exactly match the parent query.

Since the same deterministic sorting was already being appended to the parent query, I didn't go down the paths discussed in #26808 of checking whether existing sorts on the query are already perfectly deterministic.

Fixes #26808

(Apologies for not asking for permission to work on this first - this started out as an exploration of whether I could even figure out how to approach fixing an issue like this).

### CODE REVIEW

**Review by roji — APPROVED** [review]
Thanks for this PR @ascott18, and sorry it took so long to get reviewed.

This looks correct to me, and a good fix; I'm not sure why the identifier orderings were added late (after the pushdown) - hopefully that was just an oversight and not for a good reason (this area of the code was written by a team member who has left since, and it's one place where we don't have perfect expertise at the moment).

I'm good with merging this for 9.0 (though see the comment about removing the dead warning). @maumar @cincuranet would you mind giving this a look in case I've missed something?

File: test/EFCore.SqlServer.FunctionalTests/Query/NorthwindSplitIncludeNoTrackingQuerySqlServerTest.cs [hunk 1]
```diff
@@ -24,14 +24,64 @@ public virtual void Check_all_tests_overridden()
         => TestHelpers.AssertAllMethodsOverridden(GetType());
 
     public override async Task Include_collection_skip_take_no_order_by(bool async)
-        => Assert.Equal(
-            SqlServerStrings.SplitQueryOffsetWithoutOrderBy,
```
[the remark this answers is not in the document]
Comment (ascott18) [reply test/EFCore.SqlServer.FunctionalTests/Query/NorthwindSplitIncludeNoTrackingQuerySqlServerTest.cs:28]: Done!

I did also noticed that in the docs, this https://github.com/dotnet/EntityFramework.Docs/blob/main/entity-framework/core/querying/single-split-queries.md?plain=1#L109-L110 warning won't be applicable anymore. Not sure how warnings like that are usually handled when they only apply to some versions.

**Comment by ascott18** [comment]
@roji Friendly ping - is this still on track for EF 9?

**Comment by roji** [comment]
@ascott18 unfortunately not - EF 9 is already locked down and nothing but critical bug fixes is being merged in... My apologies that this took so long to process, the recent couple of months have simply been very busy. I'll make sure this gets merged in for 10.

**Comment by roji** [comment]
@ascott18 finally merged this for 10 - thanks again for contributing this! I've also submitted https://github.com/dotnet/EntityFramework.Docs/pull/4831 to amend the doc comment you referred to, thanks for that too.

**Comment by Kation** [comment]
@roji no plan to merge to 8.0?

**Comment by roji** [comment]
@Kation not at this point. The change is in an area which is generally quite tricky/risky, and EF has had this behavior since split query was introduced a long time ago. The recommendation is to make your query deterministic (i.e. specify unambigious ordering in your LINQ query).

**Pull request merged**